gutils/fsys.c

```diff
@@ -552,6 +553,10 @@ char *_GFile_find_program_dir(char *prog) {
 		program_dir = copy(path);
 	}
     }
+	if ( (buf = realpath(program_dir, NULL)) ) {
+		if ( program_dir!=NULL ) free(program_dir);
```

Comment:
You don't need a NULL check when calling `free()`.

Reply:
Addressed.